plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2020,11 +2022,10 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
    if (!chain_plug->accept_transactions())
       return start_block_result::waiting_for_block;
 
-   abort_block();
-
    chain.maybe_switch_forks([this](const transaction_metadata_ptr& trx) { _unapplied_transactions.add_forked(trx); },
                             [this](const transaction_id_type& id) { return _unapplied_transactions.get_trx(id); });
 
+   abort_block(); // does reset `_time_tracker`, so should be after `maybe_switch_forks`
```

Comment:
I think the `abort_block()` should be put back where it was. `maybe_switch_forks` EOS_ASSERTs that there is not a pending block. There are a few places where `abort_block()` is not called before `schedule_production_loop`. If we cleaned that up the `abort_block()` would not be needed at all. Regardless you would want to reset the timer instead then as `abort_block()` would not be needed.

Reply:
https://github.com/AntelopeIO/spring/pull/881/commits/b776b9d4462b24e681ef5a93938bd06c1ac1a3cb